src/Connection.php

```diff
@@ -50,20 +50,24 @@ public function __construct($resource, LoopInterface $loop)
         // See https://bugs.php.net/bug.php?id=65137
         // https://bugs.php.net/bug.php?id=41631
         // https://github.com/reactphp/socket-client/issues/24
-        $clearCompleteBuffer = (version_compare(PHP_VERSION, '5.6.8', '<'));
-
-        // @codeCoverageIgnoreStart
-        if (class_exists('React\Stream\Stream')) {
-            // legacy react/stream < 0.7 requires additional buffer property
-            $this->input = new Stream($resource, $loop);
-            if ($clearCompleteBuffer) {
-                $this->input->bufferSize = null;
-            }
-        } else {
-            // preferred react/stream >= 0.7 accepts buffer parameter
-            $this->input = new DuplexResourceStream($resource, $loop, $clearCompleteBuffer ? -1 : null);
-        }
-        // @codeCoverageIgnoreEnd
+        $clearCompleteBuffer = PHP_VERSION_ID < 50608;
+
+        // PHP < 7.1.14 (and PHP < 7.0.18) suffers from a bug when writing big
+        // chunks of data over TLS streams at once.
+        // We try to work around this by limiting the write chunk size to 8192
+        // bytes for older PHP versions only.
+        // This is only a work-around and has a noticable performance penalty on
+        // affected versions. Please update your PHP version.
+        // This applies to all streams because TLS may be enabled later on.
+        // See https://github.com/reactphp/socket/issues/105
+        $limitWriteChunks = (PHP_VERSION_ID < 70018 || (PHP_VERSION_ID < 70114 && PHP_VERSION_ID >= 70100));
```

Comment:
That should be `PHP_VERSION_ID >= 70100 && PHP_VERSION_ID < 70114` for better readability IMO.

Reply:
Unfortunately this does not match the same version range. The fix landed in PHP 7.1.14 *and* PHP 7.0.18, which means that all earlier versions are affected (such as 7.1.13 and 7.0.17, but *not* 7.0.19)

Reply:
Yes, sure, just talking about the second condition being switched.

`PHP_VERSION_ID < 70018 || PHP_VERSION_ID >= 70100 && PHP_VERSION_ID < 70114`

Reply:
I don't think either version is any clearer, so I've just updated this to match your suggestion :shipit: :+1: